src/OpenTelemetry.Resources.Gcp/GcpResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.Internal;
+using OpenTelemetry.Resources.Gcp;
+
+namespace OpenTelemetry.Resources;
```

Comment:
Should extensions go into the namespace of the thing they are extending or somewhere else?

We have had debates about this in the past so I thought I would drop a comment here to make it known this was an intentional decision.

[.NET guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/extension-methods) say this:

![image](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/assets/28367120/ec89535f-0df9-466c-b74a-955d51109dd2)

I would argue these packages/extensions are all about dependency management for users so this gets a 👍 from me.

Reply:
Most of the instrumentations put their extension methods in the namespace for the type they're extending (ex. `OpenTelemetry.Trace`, `OpenTelemetry.Metrics`), so would fit unless the other extension methods are going to be moved.